### PR TITLE
Strip password when reading from file

### PR DESCRIPTION
**Pull request by wild-endeavor**
# TL;DR
Strip the password when reading from a file.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
This is tangentially possibly related to https://github.com/flyteorg/flyte/issues/2849.  Found this nit when investigating that issue, but not sure if this is what the user was experiencing or if there is a different bug somewhere.

## Tracking Issue
https://github.com/flyteorg/flyte/issues/2849

### CODE REVIEW

**Comment by wild-endeavor** [comment]
closing see https://github.com/flyteorg/flytekit/pull/1163
